pubnub-chat-impl/src/commonTest/kotlin/com/pubnub/integration/ChatIntegrationTest.kt

```diff
@@ -738,6 +771,216 @@ class ChatIntegrationTest : BaseChatIntegrationTest() {
+        chat.initialize().await()
+        val firstResult = chat.createDirectConversation(someUser).await()
+        val firstChannelId = firstResult.channel.id
+
+        // when - create direct conversation again with same user
+        val secondResult = chat.createDirectConversation(someUser).await()
+        val secondChannelId = secondResult.channel.id
+
+        // then - should return the same channel ID (idempotent operation)
+        assertEquals(firstChannelId, secondChannelId, "Creating direct conversation twice should return same channel")
+
+        // verify memberships are consistent
+        assertEquals(
+            chat.currentUser.asImpl().copy(updated = null, lastActiveTimestamp = null),
+            secondResult.hostMembership.user.asImpl().copy(updated = null, lastActiveTimestamp = null)
+        )
+        assertEquals(someUser, secondResult.inviteeMembership.user.asImpl())
+        assertEquals(secondResult.channel, secondResult.hostMembership.channel)
+        assertEquals(secondResult.channel, secondResult.inviteeMembership.channel)
+
+        // verify no duplicate channels created - fetch channel directly
+        val fetchedChannel = chat.getChannel(firstChannelId).await()
+        assertNotNull(fetchedChannel, "Original channel should still exist")
+        assertEquals(firstChannelId, fetchedChannel.id)
+    }
+
+    @Test
+    fun isPresent_withStateTransitions_shouldReflectCorrectState() = runTest {
+        // given - create a test channel
+        val testChannelId = randomString()
+        val testChannel = chat.createChannel(testChannelId).await()
+
+        // initially, current user should not be present (not subscribed)
+        val initiallyPresent = chat.isPresent(chat.currentUser.id, testChannelId).await()
+        assertFalse(initiallyPresent, "User should not be present before subscribing")
+
+        // when - user subscribes to the channel
+        pubnub.test(backgroundScope, checkAllEvents = false) {
+            var subscription: AutoCloseable? = null
+            pubnub.awaitSubscribe(listOf(testChannelId)) {
+                subscription = testChannel.connect { }
+            }
+
+            // wait for presence to update
+            delayInMillis(2000)
+
+            // then - user should be present
+            val nowPresent = chat.isPresent(chat.currentUser.id, testChannelId).await()
+            assertTrue(nowPresent, "User should be present after subscribing")
+
+            // when - user unsubscribes
+            subscription?.close()
+
+            // wait for presence to update
+            delayInMillis(2000)
+
+            // then - user should not be present anymore
+            val afterUnsubscribe = chat.isPresent(chat.currentUser.id, testChannelId).await()
+            assertFalse(afterUnsubscribe, "User should not be present after unsubscribing")
+        }
+    }
+
+    @Test
+    fun whoIsPresent_withMultipleUsers_shouldReturnAllPresent() = runTest {
```

Comment:
Does the existing `canCheck_whoIsPresent()` test cover most of the behaviors you are testing below?